internal/pkg/agent/install/uninstall.go

```diff
@@ -290,3 +297,36 @@ func applyDynamics(ctx context.Context, log *logger.Logger, cfg *config.Config)
 
 	return config.NewConfigFrom(finalConfig)
 }
+
+// killWatcher finds and kills any running Elastic Agent watcher.
+func killWatcher() error {
+	for {
+		// finding and killing watchers is performed in a loop until no
+		// more watchers are existing, this ensures that during uninstall
+		// that no matter what the watchers are dead before going any further
+		pids, errs := utils.GetWatcherPIDs()
+		if errs != nil {
+			return errs
```

Comment:
Even if it'd return several errors, as they aren't individually handled here, I'd suggest to stick to the convention
```suggestion
		pids, err := utils.GetWatcherPIDs()
		if errs != nil {
			return err
```